Mathlib/Computability/QueryComplexity/Basic.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2025 Tomaz Mascarenhas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving, Tomaz Mascarenhas
+-/
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Algebra.Group.Nat.Defs
+import Mathlib.Computability.QueryComplexity.Defs
+import Mathlib.Tactic.Cases
+
+/-!
+## Basic properties of `Comp` and its Monad instance.
+-/
+
+variable {I : Type*}
+variable {ι : I → Type*}
+variable {ω : {i : I} → ι i → Type*}
+variable {s t u : Set I}
+variable {α γ : Type*}
+variable {β : Type _}
+
+namespace QueryComplexity
+
+namespace Comp
+
+/-- The definition of `Comp.map` -/
+lemma map_eq (f : α → β) (x : Comp ι ω s α) : f <$> x = x >>= (fun x ↦ pure (f x)) := rfl
+
+/-- The definition of `Comp.bind` as `simp` lemmas -/
+@[simp]
+lemma pure'_bind (x : α) (f : α → Comp ι ω s β) : (pure' x : Comp ι ω s α) >>= f = f x := rfl
```

Comment:
Usually it's a bad idea to have two ways of spelling things. Can you add a simp lemma that `pure' = pure`, and then remove all the `pure'` theorems from this file?